src/classLibrary/Utility/FileFactory.ts

```diff
@@ -0,0 +1,112 @@
+import fs from "fs";
+import path from "path";
+import { isBinaryFile } from 'isbinaryfile';
+import { DirectorySystemInstance, 
+    FileContent, 
+    FileSystemInstance, 
+    GlobSyncOptions, 
+    IFileFactory, 
+    IFileSystemInstanceMetadata, 
+    Ignore, 
+    VirtualFileSystemInstance } from "../interfaces/interfaces";
+import _glob from "glob";
+
+class FileFactory implements IFileFactory {
+    readonly FromRoot: string;
+    readonly ToRoot: string;
+
+    constructor(fromRoot: string, toRoot: string) {
+        this.FromRoot = fromRoot;
+        this.ToRoot = toRoot;
+    }
+
+     /**
+	 * Generates the file instance according to the relative path to the file/directory, directory from the source root path, destination root path
+	 * returns the file
+	 * @param relativePath relative path from the source root path including the optional filepaths
+     * @param from Optional filepath of the file/directory from the source root filepath
+     * @param to Optional filepath of the file/directory from the destination root filepath
+     * @param to Optional ignore for the resolvers to ignore content or filepath
+	 */
+    CreateFileSystemInstance(relativePath:string, from: string = "./", to: string = './', ignore?: Ignore): VirtualFileSystemInstance {
+        let absFrom = path.resolve(this.FromRoot, from, relativePath);
+        let absTo = path.resolve(this.ToRoot, to, relativePath);
+        let fileMetadata: IFileSystemInstanceMetadata = {
+            relativePath: relativePath,
+            sourceAbsolutePath: absFrom,
+            destinationAbsolutePath: absTo,
+        };
+        let ignoreConfig: Ignore = {
+            variableResolver:  {},
+            inlineResolver: {},
+            ifElseResolver: {},
+            guidResolver: {},
+            custom: {}
+        }
+        if (ignore !== undefined) {
+            ignoreConfig = ignore;
+        }
```

Comment:
I think for this portion, we can use lodash [`defaultsDeep` ](https://lodash.com/docs/4.17.15#defaultsDeep)